extensions/poetry/tests.ts

```diff
@@ -0,0 +1,93 @@
+//
+// Run this with `deno run --allow-all tests.ts`.
+//
+// The extension has the capability of running external commands (`poetry`) and
+// change the working directory it is executed in.
+//
+// This file is not meant as a comprehensive integration test suite, but rather
+// it is intended to provide an automated way of copying test fixtures to a
+// temporary directory, run the extension there, and subsequently clean up.
+//
+
+import { describe, afterAll, beforeAll, it, } from "https://deno.land/std@0.148.0/testing/bdd.ts"
+import { copy } from "https://deno.land/std@0.148.0/fs/mod.ts"
+import { assert } from "https://deno.land/std@0.148.0/testing/asserts.ts"
+
+class Phylum {
+  readonly xdgDataHome: string
+  readonly fixturesPath: string
+
+  constructor (tempDir: string) {
+    this.xdgDataHome = tempDir
+    this.fixturesPath = tempDir + '/fixtures'
+  }
+
+  async run (args: string[], cwd?: string) {
+    let process = Deno.run({
+      cmd: ['phylum', ...args],
+      env: { 'XDG_DATA_HOME': await this.xdgDataHome },
+      cwd,
+      stdout: 'inherit',
+      stderr: 'inherit',
+    })
+
+    const status = await process.status()
+    await process.close()
+
+    return status
+  }
+
+  async installExtension (extension: string) {
+    await this.run(['extension', 'install', '-y', extension])
+  }
```

Comment:
I wonder, do we actually need to test the extension through `phylum` here? If the goal is just to ensure the extension itself is working, maybe executing the extension's typescript code directly without `phylum` makes more sense? It probably would have less boilerplate at least.

Generally as an extension developer I would expect the installation and arg passing process to be tested by our CLI itself, so no reason to test it with the extension?

Reply:
We do. We cannot import `phylum` as a module as it is injected by our own runtime and not available anywhere else.

This is a lot of boilerplate indeed, but it could be trimmed down significantly by #551.